crates/binjs_io/src/io/content.rs

```diff
@@ -0,0 +1,120 @@
+    pub interface_names: T,
+    pub string_literals: T,
+    pub list_lengths: T,
+}
+impl<T> ContentInfo<T> {
+    /// Initialize a new `ContentInfo`.
+    pub fn with<F>(f: F) -> Self
+        where F: Fn(&str) -> T
+    {
+        ContentInfo {
+            bools: f("bools"),
+            floats: f("floats"),
+			unsigned_longs: f("unsigned_longs"),
+			string_enums: f("string_enums"),
+			property_keys: f("property_keys"),
+			identifier_names: f("identifier_names"),
+			interface_names: f("interface_names"),
+			string_literals: f("string_literals"),
+			list_lengths: f("list_lengths"),
+        }
+    }
+
+    /// Convert a `ContentInfo` into another one.
+    pub fn into_with<F, U>(self, f: F) -> ContentInfo<U>
+        where F: Fn(T, &str) -> U
+    {
+        ContentInfo {
+			bools: f(self.bools, "bools"),
+			floats: f(self.floats, "floats"),
+			unsigned_longs: f(self.unsigned_longs, "unsigned_longs"),
+			string_enums: f(self.string_enums, "string_enums"),
+			property_keys: f(self.property_keys, "property_keys"),
+			identifier_names: f(self.identifier_names, "identifier_names"),
+			interface_names: f(self.interface_names, "interface_names"),
+			string_literals: f(self.string_literals, "string_literals"),
+			list_lengths: f(self.list_lengths, "list_lengths"),
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (&T, &'static str)> {
+        vec![
+            (&self.bools, "bools"),
+            (&self.floats, "floats"),
+			(&self.unsigned_longs, "unsigned_longs"),
+			(&self.string_enums, "string_enums"),
+			(&self.property_keys, "property_keys"),
+			(&self.identifier_names, "identifier_names"),
+			(&self.interface_names, "interface_names"),
+			(&self.string_literals, "string_literals"),
+			(&self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = (&mut T, &'static str)> {
+        vec![
+            (&mut self.bools, "bools"),
+            (&mut self.floats, "floats"),
+			(&mut self.unsigned_longs, "unsigned_longs"),
+			(&mut self.string_enums, "string_enums"),
+			(&mut self.property_keys, "property_keys"),
+			(&mut self.identifier_names, "identifier_names"),
+			(&mut self.interface_names, "interface_names"),
+			(&mut self.string_literals, "string_literals"),
+			(&mut self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+
+    pub fn into_iter(self) -> impl Iterator<Item = (T, &'static str)> {
+        vec![
+            (self.bools, "bools"),
+            (self.floats, "floats"),
+			(self.unsigned_longs, "unsigned_longs"),
+			(self.string_enums, "string_enums"),
+			(self.property_keys, "property_keys"),
+			(self.identifier_names, "identifier_names"),
+			(self.interface_names, "interface_names"),
+			(self.string_literals, "string_literals"),
+			(self.list_lengths, "list_lengths"),
+        ].into_iter()
+    }
+}
+
+impl<T> std::iter::FromIterator<(T, &'static str)> for ContentInfo<Option<T>> {
```

Comment:
where is this used, and what is this for?

Reply:
It deserializes a `ContentInfo` from a data structure. So, typical use

```rust
let mut info = HashMap::new();
// Somehow, fill `info` e.g. by reading a dictionary or a compressed file.

let info2: ContentInfo<_> = info.into_iter()
    .collect();
// Optionally, call `into_with` to convert from a `ContentInfo<Option<T>>` into a `ContentInfo<T>`.

```